docs: (IAC-372) Updated the commands related to Service Principal creation

riragh opened the pull request:
## Changes:
Some of the commands in the `AzureHelpTopics.md` document were outdated. Updated the commands related to Service Principal creation.

riragh commented:
For az CLI version `2.37.0` or below, the command `az ad sp credential reset` requires `--name` argument.

While working on FIPS changes I updated the az CLI version on my machine to `2.45.0` which has required argument changed from --name to --id, hence I updated the command to use the --display-name instead. To keep consistency with the az version specified in main branch, I'll revert this command back to original.

dhoucgitter commented:
> For az CLI version `2.37.0` or below, the command `az ad sp credential reset` requires `--name` argument.
> 
> While working on FIPS changes I updated the az CLI version on my machine to `2.45.0` which has required argument changed from --name to --id, hence I updated the command to use the --display-name instead. To keep consistency with the az version specified in main branch, I'll revert this command back to original.

Just trying to understand, so the commands forms in this PR will work with az cli v2.24.2 mentioned in main README.md, but not az cli 2.45.0?

riragh commented:
> > For az CLI version `2.37.0` or below, the command `az ad sp credential reset` requires `--name` argument.
> > While working on FIPS changes I updated the az CLI version on my machine to `2.45.0` which has required argument changed from --name to --id, hence I updated the command to use the --display-name instead. To keep consistency with the az version specified in main branch, I'll revert this command back to original.
> 
> Just trying to understand, so the command forms in this PR will work with az cli v2.24.2 mentioned in main README.md, but not az cli 2.45.0?

Yes this command works for all the versions below 2.37.0.
